app/actions/websocket/channel.test.ts

```diff
@@ -0,0 +1,577 @@
+
+import type ServerDataOperator from '@database/operator/server_data_operator';
+
+jest.mock('@database/manager');
+jest.mock('@store/ephemeral_store');
+jest.mock('@actions/local/category');
+jest.mock('@actions/remote/channel');
+jest.mock('@actions/remote/role');
+jest.mock('@actions/remote/post');
+jest.mock('@actions/remote/user');
+jest.mock('@actions/local/channel');
+jest.mock('@actions/local/post');
+jest.mock('@queries/servers/channel');
+jest.mock('@queries/servers/system');
+jest.mock('@queries/servers/user');
+jest.mock('@calls/actions');
+jest.mock('@actions/local/user');
+jest.mock('@utils/log');
+jest.mock('@calls/state');
+jest.mock('@calls/actions/calls');
+
+const serverUrl = 'baseHandler.test.com';
+
+describe('WebSocket Channel Actions', () => {
+    let msg: any;
+    let operator: ServerDataOperator;
+
+    const teamId = 'teamid1';
+    const channelId = 'channelid1';
+    const userId = 'userid1';
+
+    beforeEach(async () => {
+        await DatabaseManager.init([serverUrl]);
+        operator = DatabaseManager.serverDatabases[serverUrl]!.operator;
+        msg = {
+            data: {
+                team_id: teamId,
+                channel_id: channelId,
+                channel: JSON.stringify({id: channelId, type: General.PRIVATE_CHANNEL, team_id: teamId}),
+            },
+            broadcast: {
+                channel_id: channelId,
+                userId,
+                user_id: userId,
+            },
+        };
+        DatabaseManager.getActiveServerUrl = jest.fn().mockResolvedValue(serverUrl);
+    });
+
+    afterEach(async () => {
+        await DatabaseManager.destroyServerDatabase(serverUrl);
+        jest.restoreAllMocks();
+    });
+
+    describe('handleChannelCreatedEvent', () => {
+        it('should handle channel created event', async () => {
+            (EphemeralStore.creatingChannel as boolean) = false;
+            (getChannelById as jest.Mock).mockResolvedValue(null);
+            (fetchMyChannel as jest.Mock).mockResolvedValue({channels: [{}], memberships: [{}]});
+            (prepareMyChannelsForTeam as jest.Mock).mockResolvedValue([[]]);
+            (addChannelToDefaultCategory as jest.Mock).mockResolvedValue({models: []});
+
+            await handleChannelCreatedEvent(serverUrl, msg);
+
+            expect(getChannelById).toHaveBeenCalled();
+            expect(fetchMyChannel).toHaveBeenCalledWith(serverUrl, teamId, channelId, true);
+            expect(prepareMyChannelsForTeam).toHaveBeenCalled();
+            expect(addChannelToDefaultCategory).toHaveBeenCalled();
+        });
+
+        it('should handle channel created event - no channels', async () => {
+            (EphemeralStore.creatingChannel as boolean) = false;
+            (getChannelById as jest.Mock).mockResolvedValue(null);
+            (fetchMyChannel as jest.Mock).mockResolvedValue({channels: null, memberships: [{}]});
```

Comment:
fetch my channel will never return null in `channels`. What it can return is an object `{error: 'some error'}`, but not an object like this.